tensorflow_serving/model_servers/main.cc

```diff
@@ -276,6 +276,17 @@ void RunServer(int port, std::unique_ptr<ServerCore> core,
   builder.RegisterService(&model_service);
   builder.RegisterService(&prediction_service);
   builder.SetMaxMessageSize(tensorflow::kint32max);
+  const std::vector<string> channel_arguments =
```

Comment:
Could you move the flag parsing to a separate helper function please. Then have a single callsite to builder.AddChannelArgument().

Reply:
I'm happy to move the parsing, but I don't see how I can achieve a single callsite since it has two signatures: https://github.com/grpc/grpc/blob/master/include/grpc%2B%2B/impl/channel_argument_option.h

I'm not very experienced with cpp, thanks for your help!

Reply:
Oh, I see. I missed that. Please add some comments explaining what you are doing, since it's somewhat unusual. I gather that you want to handle a variety of arguments, some of which gRPC expects to have int values and others that will have string values, and you are hoping that you can distinguish based on int parsability.

The ideal approach would be to query gRPC for a given flag to find out its expected type (int or string), but assuming that isn't possible then what you are doing seems sensible. Please do document.